File: 908-middle-of-the-linked-list/middle-of-the-linked-list.py
class Solution:
    def middleNode(self, no_cabeca_da_lista: Optional[ListNode]) -> Optional[ListNode]:
        ponteiro_lento = no_cabeca_da_lista
        ponteiro_rapido = no_cabeca_da_lista

        while ponteiro_rapido is not None and ponteiro_rapido.next is not None:
            ponteiro_lento = ponteiro_lento.next
            ponteiro_rapido = ponteiro_rapido.next.next
        
        return ponteiro_lento

File: CSharp/Easy/908-middle-of-the-linked-list/middle-of-the-linked-list.cs
public class Solution {
    public ListNode MiddleNode(ListNode no_cabeca_da_lista) {
        ListNode ponteiro_lento = no_cabeca_da_lista;
        ListNode ponteiro_rapido = no_cabeca_da_lista;

        while (ponteiro_rapido != null && ponteiro_rapido.next != null){
            ponteiro_lento = ponteiro_lento.next;
            ponteiro_rapido = ponteiro_rapido.next.next;
        }
        
        return ponteiro_lento;
    }
}